src/bokeh/models/ui/menus.py

```diff
@@ -70,7 +70,17 @@ class ActionItem(MenuItem):
     def __init__(self, *args, **kwargs) -> None:
         super().__init__(*args, **kwargs)
 
-    icon = Nullable(Either(Image, Enum(ToolIcon), Regex(r"^--"), Regex(r"^\.")), help="""
+    checked = Nullable(Bool, default=None, help="""
+    Whether an item is marked as checked/active.
+
+    Checked item is represented with a tick mark on the left hand side
+    of an item. Unchecked item is represented with an empty space.
+
+    The menu will allocate a column for check marks for all its items if
+    at least one item has set a boolean value for ``checked`` property.
```

Comment:
This seems a little awkward. If I want a list of checkable, but all initially unchecked items, I need to either:
* pick one at random to set `checked=False` (arbitrary, inconsistent)
* explicitly set all `checked=False` (a bit tedious, False is a reasonable default to assume for checked)

?

Reply:
Alternatives are:
1. Leave `CheckableItem` undeprecated and override `checked` default value with `False`.
2. Add `Menu.checkable: bool | "auto"`, where `"auto"` is the default current behavior. `False` seems not particlarly useful though.

Reply:
I'm not sure I like any of these. It just seems to me that "checkability" is an explicit up-front decision, so there's no need for all the magic on the base class (i.e. `ActionItem` should just not have `checked at all). But if you don't like that route for some reason I suppose the way things is preferable over the two options above.

Reply:
Given that non of these properties are mutually exclusive, I figured that only having `MenuItem` will be preferable, instead of more complex complex model hierarchy, and easier for the user if they change their mind (no need to change the model and imports). Other frameworks that provide context menus (e.g. Qt, WinForms) do the same.

Reply:
Continuation:

- https://github.com/bokeh/bokeh/discussions/14338)